src/domains/migration/hooks/use-migration-transactions.ts

```diff
@@ -49,20 +48,7 @@ export const useMigrationTransactions = ({ profile }: { profile: Contracts.IProf
 	}, [profileIsRestoring]);
 
 	const migrationTransactions = useMemo(
-		() =>
-			latestTransactions.filter((transaction) => {
-				const polygonAddress = transaction.memo();
-
-				if (!polygonAddress) {
-					return false;
-				}
-
-				if (transaction.recipient() !== migrationWalletAddress()) {
-					return false;
-				}
-
-				return ethers.utils.isAddress(polygonAddress);
-			}),
+		() => latestTransactions.filter((transaction) => isValidMigrationTransaction(transaction)),
```

Comment:
this filters transactions on the ARKVault side, but won't that cause issues if you have an old valid migration and then a lot of newer invalid migration transactions for an address as it will filter them all out while missing the valid migration on the initial page?

More concrete example:
- address has 101 transactions, the oldest transaction is a valid migration and the 100 other ones are not
- you fetch the newest 100 transactions, all are removed locally because invalid
- there's still 1 transaction left, but it wasn't included in the API call so it's not shown, while that one was the valid transaction

Instead of applying that logic locally, I'd have it apply to the initial query to fetch transactions so you get a response with valid transactions for the overview to avoid scenarios like above 🤔

Reply:
@ItsANameToo makes totally sense. Missed the supported query params (`fee.from` & `amount.from`) in the transaction API. I added those now and should be working ok.

Mentioning here also that the only client-side validation that can't be included in the query is the validation of the polygon address (vendorField), so we should now be as close as possible.

Thanks for pointing out :100: